src/duke/Duke.java

```diff
@@ -0,0 +1,115 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Duke is an interactive chat-bot that allows the user to
+ * add, modify or delete tasks from a to-do list, which is
+ * saved locally as a .txt file.
+ * @author Ng Weng Fai
+ */
+public class Duke {
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Instantiates Duke, which loads data from a previously
+     * saved  duke.txt, if it exists.
+     * @param filepath file path of duke.txt.
+     */
+    public Duke(String filepath) {
+        taskList = TaskList.startList();
+        storage = new Storage(filepath);
+        try {
+            storage.populateList(taskList);
+        } catch (Exception e) {
+            Ui.displayMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Starts up Duke.
+     */
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        Ui.displayMessage("Hello! I'm duke.Duke\n\tWhat can I do for you?");
+        String input;
+        String msg;
+        boolean cont = true;
```

Comment:
Again, try not to shorten variable names. For boolean variable names, name them in the format is____ instead